src/System Application/App/AI/src/Azure OpenAI/Chat Completion/AOAIToolsImpl.Codeunit.al

```diff
@@ -47,40 +90,83 @@ codeunit 7778 "AOAI Tools Impl"
         Tools.RemoveAt(Id);
     end;
 
-    procedure ClearTools()
-    begin
-        Clear(Tools);
-    end;
-
     [NonDebuggable]
+    [Obsolete('Use GetTool() that takes in a function name and var for AOAI Function interface.', '25.0')]
     procedure GetTools(): List of [JsonObject]
     begin
         exit(Tools);
     end;
+#endif
+
+    procedure DeleteTool(Name: Text): Boolean
+    var
+        Index: Integer;
+    begin
+        if not FunctionNames.ContainsKey(Name) then
+            exit(false);
+
+        Index := FunctionNames.get(Name);
+        FunctionNames.Remove(Name);
+
+        for Index := Index to FunctionNames.Count() do begin
```

Comment:
Just double check:
It will never cause OutOfIndex error because the `Index` will not reach `FunctionNames.Count()` right?